tests/storage/hotplug.go

```diff
@@ -1019,10 +1026,91 @@ var _ = SIGDescribe("Hotplug", func() {
 				addVolumeFunc(vmi.Name, vmi.Namespace, volumeName, dv.Name, v1.DiskBusSCSI, false, "")
 				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
-				VerifyVolumeAndDiskVMIAdded(virtClient, vmi, volumeName)
+				verifyVolumeAndDiskVMIAdded(virtClient, vmi, volumeName)
 				verifyVolumeStatus(vmi, v1.VolumeReady, "", volumeName)
 			})
 		})
+
+		Context("disk mutating sidecar", func() {
+			const (
+				hookSidecarImage     = "example-disk-mutation-hook-sidecar"
+				sidecarContainerName = "hook-sidecar-0"
+				newDiskImgName       = "kubevirt-disk.img"
+			)
+
+			var (
+				vm *v1.VirtualMachine
+				dv *cdiv1.DataVolume
+				sc string
+			)
+
+			renderSidecar := func() map[string]string {
+				return map[string]string{
+					"hooks.kubevirt.io/hookSidecars":         fmt.Sprintf(`[{"image": "%s/%s:%s", "imagePullPolicy": "IfNotPresent"}]`, flags.KubeVirtUtilityRepoPrefix, hookSidecarImage, flags.KubeVirtUtilityVersionTag),
+					"diskimage.vm.kubevirt.io/bootImageName": newDiskImgName,
+				}
+			}
+
+			BeforeEach(func() {
+				if !libstorage.HasCDI() {
+					Skip("Skip tests when CDI is not present")
+				}
+				var exists bool
+				sc, exists = libstorage.GetRWXBlockStorageClass()
+				if !exists {
+					Skip("Skip test when RWXBlock storage class is not present")
+				}
+			})
+
+			AfterEach(func() {
+				if vm != nil {
+					err := virtClient.VirtualMachine(vm.Namespace).Delete(vm.Name, &metav1.DeleteOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					vm = nil
+				}
+				libstorage.DeleteDataVolume(&dv)
+			})
+
+			DescribeTable("should be able to add and remove volumes", func(addVolumeFunc addVolumeFunction, removeVolumeFunc removeVolumeFunction, volumeMode corev1.PersistentVolumeMode, vmiOnly bool) {
+				var err error
+				url := cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskCirros)
+				dv = libstorage.NewDataVolumeWithRegistryImport(url, util.NamespaceTestDefault, k8sv1.ReadWriteMany)
```

Comment:
Don't think we need RWX, we make no use of that in the test.

Reply:
I tried using `k8sv1.ReadWriteOnce` initially and I remember seeing some error in the pipeline. Let me try changing it back and see what I get there. 

On a side note, do you have some pointers on how I can run these tests locally? If you don't mind could you please share them with me. It takes too long for me to run these tests on CI and verify.

Reply:
So for development there is kubevirtci, which is basically a cluster made of VMs that runs on your local machine. You will need nested virtualization enabled on your machine. It should be as simple as running `make cluster-up` this will bring up a cluster with 1 node running k8s-1.23 by default. Then run `make cluster-sync` this will compile and deploy kubevirt on that cluster. Then running `make functest` will execute all functional tests on that cluster, which takes forever, but you can manually focus on a particular test by adding an `F` in front if. So if your test is an `It`, you can focus it by changing it to `FIt`, Same with describe and tables. Before pushing remember to remove the focus from the test.

Reply:
If you want to run rook-ceph storage, before running `make cluster-up` you have to `export KUBEVIRT_STORAGE=rook-ceph-default` and it will deploy ceph storage in your cluster.

Reply:
Thank you soo much for the pointers. I was just running `make cluster-up` earlier and the new tests were always skipped. Setting `KUBEVIRT_STORAGE=rook-ceph-default` and `KUBEVIRT_DEPLOY_NFS_CSI=true` did the trick for me and it actually saves a lot of time in debugging :)

Reply:
I am kind of surprised this works. You pass in RWOBlock storage class function, and on this line you create a RWX volume. I guess if the storage class just happens to also allow RWX (like rook-ceph) you will be fine. But for others that only have RWOBlock (topolvm for instance) I think this test will fail. If it important to have RWX, pass in an RWX storage func.

Reply:
Sorry, I don't follow your comment complete.

If I understand it correctly, this line only deals with creating the boot volume that always uses RWXFilesystem based on `libstorage.NewDataVolumeWithRegistryImport` implementation. The `storageClassFunc` passed to the test is only used for creating the hotplug volumes, and this happens in the `verifyAttachDetachVolume` helper function. Please correctly me if I am missing something here.

Reply:
I am sorry you are right, I thought we were passing in the storage class into this function, but we are not.